### ISSUE TITLE
Unable to execute `MsgVoteWeighted` using the stargate client 

### ISSUE THREAD

**Issue by spoo-bar** [cosmos/cosmjs#1223]
I am trying to do weighted voting via the cosmjs library stargate client. 
When I pass in this JSON input, i get the error given below.
```json
{
    "typeUrl": "/cosmos.gov.v1beta1.MsgVoteWeighted",
    "value": {
        "proposalId": "4",
        "voter": "juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y",
        "options": [
            {
                "option": "VOTE_OPTION_YES",
                "weight": "0.7"
            },
            {
                "option": "VOTE_OPTION_NO",
                "weight": "0.1"
            },
            {
                "option": "VOTE_OPTION_NO_WITH_VETO",
                "weight": "0.1"
            },
            {
                "option": "VOTE_OPTION_ABSTAIN",
                "weight": "0.1"
            }
        ]
    }
}
```

```
Error: Query failed with (18): 
github.com/cosmos/cosmos-sdk/baseapp.gRPCErrorToSDKError
	github.com/cosmos/cosmos-sdk@v0.45.4/baseapp/abci.go:590
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).handleQueryGRPC
	github.com/cosmos/cosmos-sdk@v0.45.4/baseapp/abci.go:579
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Query
	github.com/cosmos/cosmos-sdk@v0.45.4/baseapp/abci.go:421
github.com/tendermint/tendermint/abci/client.(*localClient).QuerySync
	github.com/tendermint/tendermint@v0.34.19/abci/client/local_client.go:256
github.com/tendermint/tendermint/proxy.(*appConnQuery).QuerySync
	github.com/tendermint/tendermint@v0.34.19/proxy/app_conn.go:159
github.com/tendermint/tendermint/rpc/core.ABCIQuery
	github.com/tendermint/tendermint@v0.34.19/rpc/core/abci.go:20
reflect.Value.call
	reflect/value.go:556
reflect.Value.Call
	reflect/value.go:339
github.com/tendermint/tendermint/rpc/jsonrpc/server.makeJSONRPCHandler.func1
	github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_json_handler.go:96
github.com/tendermint/tendermint/rpc/jsonrpc/server.handleInvalidJSONRPCPaths.func1
	github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_json_handler.go:122
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2084
net/http.(*ServeMux).ServeHTTP
	net/http/server.go:2462
github.com/tendermint/tendermint/rpc/jsonrpc/server.maxBytesHandler.ServeHTTP
	github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_server.go:236
github.com/tendermint/tendermint/rpc/jsonrpc/server.RecoverAndLogHandler.func1
	github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_server.go:209
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2084
net/http.serverHandler.ServeHTTP
	net/http/server.go:2916
net/http.(*conn).serve
	net/http/server.go:1966
math/big: cannot unmarshal "0.7" into a *big.Int: tx parse error: invalid request
    at QueryClient.queryUnverified (queryclient.js?0f47:106:1)
    at async Object.simulate (queries.js?7f6b:50:1)
    at async SigningStargateClient.simulate (signingstargateclient.js?6ad0:89:27)
    at async SigningStargateClient.signAndBroadcast (signingstargateclient.js?6ad0:160:1)
```

However, when I try to do the same via CLI, and pass on this msg, the unmarshalling happens fine and the tx passes.
```json
{
    "body": {
        "messages": [
            {
                "@type": "/cosmos.gov.v1beta1.MsgVoteWeighted",
                "proposal_id": "4",
                "voter": "juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y",
                "options": [
                    {
                        "option": "VOTE_OPTION_YES",
                        "weight": "0.700000000000000000"
                    },
                    {
                        "option": "VOTE_OPTION_NO",
                        "weight": "0.100000000000000000"
                    },
                    {
                        "option": "VOTE_OPTION_ABSTAIN",
                        "weight": "0.100000000000000000"
                    },
                    {
                        "option": "VOTE_OPTION_NO_WITH_VETO",
                        "weight": "0.100000000000000000"
                    }
                ]
            }
        ],
        "memo": "",
        "timeout_height": "0",
        "extension_options": [],
        "non_critical_extension_options": []
    },
    "auth_info": { },
    "signatures": []
}
```

I have ensured that I am registering MsgVoteWeighted as well.
```typescript
stargateClient.registry.register("/cosmos.gov.v1beta1.MsgVoteWeighted", MsgVoteWeighted)
```

Is there something else which needs to be setup to be able to handle this msg? 
Should the marshalling on the frontend side be handled in a specific way because currently I am just building the json manually.

**Comment by webmaster128** [comment]
Thanks for bringing this up. Could you have a look at https://github.com/cosmos/cosmjs/pull/1224/files, especially the `describe("MsgVoteWeighted", () => {` block in `packages/stargate/src/modules/gov/messages.spec.ts`? There you see how the `MsgVoteWeighted` needs to be constructed. Unfortunately the weight parameter is a string of atomic units in the protobuf API (i.e. "700000000000000000" is 0.7).

**Comment by webmaster128** [comment]
I think this is done when looking at https://github.com/cosmos/cosmjs/pull/1224. If there is anything left, feel free to create a new more specific follow-up issue.

**Comment by webmaster128** [comment]
Shipped as part of CosmJS 0.29.0
